cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/etl/api/StageConfigurer.java

```diff
@@ -36,6 +37,13 @@
   Schema getInputSchema();
 
   /**
+   * get the input schemas for this stage, or null if its unknown
+   * @return map of input schemas
+   */
+  @Nullable
+  Map<String, Schema> getInputSchemas();
```

Comment:
As we discussed earlier, we will need to refactor this API. Can you add the JIRA with a TODO over here?